src/apps/api/views/analytics.py

```diff
@@ -0,0 +1,72 @@
+from django.db.models import Count
+from django.contrib.auth import get_user_model
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+from competitions.models import Competition, Submission
+
+
+User = get_user_model()
+
+
+def build_request_object(model_name, filter_param, time_unit, start_date, end_date):
+    ob = {}
+    filter_args = {
+        filter_param + '__range': [start_date, end_date]
+    }
+    for date in model_name.objects.filter(**filter_args).datetimes(filter_param, time_unit):
+        if date.year not in ob:
+            ob[date.year] = {}
+        if date.month not in ob[date.year]:
+            if time_unit == 'day':
+                ob[date.year][date.month] = {}
+            else:
+                ob[date.year][date.month] = {
+                    'total': 0,
+                }
+                filter_args = {
+                    filter_param + '__month': date.month,
+                    filter_param + '__year': date.year,
+                }
+                this_month = model_name.objects.filter(**filter_args).aggregate(Count('pk'))
+                ob[date.year][date.month]['total'] = this_month['pk__count']
+
+        if time_unit == 'day':
+            if date.day not in ob[date.year][date.month]:
+                ob[date.year][date.month][date.day] = {
+                    'total': 0,
+                }
+            filter_args = {
+                filter_param + '__year': date.year,
+                filter_param + '__month': date.month,
+                filter_param + '__day': date.day,
+            }
+            this_day = model_name.objects.filter(**filter_args).aggregate(Count('pk'))
+            ob[date.year][date.month][date.day]['total'] = this_day['pk__count']
+
+    return ob
+
+
+@api_view(['GET'])
+def analytics_detail(request):
```

Comment:
how many queries does this endpoint actually make? we might need to do some optimizing here.

Reply:
Only makes 3 queries per request now.